base/show.jl

```diff
@@ -992,13 +992,34 @@ function show(io::IO, l::Core.MethodInstance)
             show(io, def)
         else
             print(io, "MethodInstance for ")
-            show_tuple_as_call(io, def.name, l.specTypes)
+            show_tuple_as_call(io, def.name, l.specTypes, false, nothing, nothing, true)
         end
     else
         print(io, "Toplevel MethodInstance thunk")
     end
 end
 
+function show(io::IO, mi_info::Core.Compiler.Timings.InferenceFrameInfo)
+    mi = mi_info.mi
+    def = mi.def
+    if isa(def, Method)
+        if isdefined(def, :generator) && mi === def.generator
+            print(io, "InferenceFrameInfo generator for ")
+            show(io, def)
+        else
+            print(io, "InferenceFrameInfo for ")
+            argnames = [isa(a, Core.Const) ? a.val : "" for a in mi_info.slottypes]
+            show_tuple_as_call(io, def.name, mi.specTypes, false, nothing, argnames, true)
+        end
+    else
+        print(io, "Toplevel InferenceFrameInfo thunk")
+    end
+end
+
+function show(io::IO, tinf::Core.Compiler.Timings.Timing)
+    print(io, "Core.Compiler.Timings.Timing(", tinf.mi_info, ") with ", length(tinf.children), " children")
+end
```

Comment:
Thanks for this! The super-long graph display was driving me crazy as well. I should have made this cleanup in the very first PR that introduced this functionality. Thank you for cleaning it up! 👍 👍